src/api.v2/model/Sample.js

```diff
@@ -16,6 +16,47 @@ class Sample extends BasicModel {
   constructor(sql = sqlClient.get()) {
     super(sql, tableNames.SAMPLE, sampleFields);
   }
+
+  async getSamples(experimentId) {
+    const { sql } = this;
+
+    const fieldsWithMetadata = [...sampleFields, sql.raw(
+      `${replaceNullsWithObject('jsonb_object_agg(key, value)', 'key')} as metadata`,
+    )];
+    const sampleFieldsWithAlias = sampleFields.map((field) => `s.${field}`);
+
+    const metadataQuery = sql.select(fieldsWithMetadata)
+      .from(sql.select([...sampleFieldsWithAlias, 'm.key', 'sm_map.value'])
+        .from({ s: tableNames.SAMPLE })
+        .leftJoin(`${tableNames.METADATA_TRACK} as m`, 's.experiment_id', 'm.experiment_id')
+        .leftJoin(`${tableNames.SAMPLE_IN_METADATA_TRACK_MAP} as sm_map`, 's.id', 'sm_map.sample_id')
+        .where('s.experiment_id', experimentId)
+        .as('mainQuery'))
+      .groupBy(sampleFields)
+      .as('select_metadata');
+
+    const sampleFileFields = ['sample_file_type', 's3_path', 'size', 'valid', 'upload_status'];
```

Comment:
`valid` was removed in the file upload PR that I merged today, so any references to `valid` will need to be removed form here and openapi yamls and all of that